test/adapters/http_server.spec.js

```diff
@@ -276,7 +276,7 @@ describe('read http_server', function() {
             return finish;
         })
         .then(function(results) {
-            expect(results.errors).to.deep.equal(['Invalid CSV data: Error: Row length does not match headers']);
+            expect(results.errors).to.deep.equal(['Invalid CSV data: Red Dragon, 24.04 at line 2']);
```

Comment:
These darn csv parser all have their own magical error messages. The new one gives me the full line on an error while the old one would indicate the message you liked `"Row length does not match headers"`. Then the messages diverge even more once certain flags are turned on and off: https://github.com/C2FO/fast-csv/blob/master/lib/parser/parser_stream.js#L127

Not sure how to easily reconcile this... I'm thinking I could make the message instead look like so:

```
Invalid CSV data: "Red Dragon, 24.04" at line 2 does not match header line "product,amount,saleTime"
```

@davidvgalbraith what do you think ? its a bit more verbose but allows the user to compare the problematic line with the header line ?